File: GotDotNet.com/ES Hands On Lab/VB/Exercise 3/HelloEnterpriseServices/HelloTx.vb
Imports System.Windows.Forms
Imports System.EnterpriseServices

<Transaction(TransactionOption.Required)> _
Public Class HelloTx
    Inherits ServicedComponent

    Public Sub SayHello(ByVal Name As String)
        Dim txId As Guid
        txId = ContextUtil.TransactionId
        Select Case MessageBox.Show(String.Format( _
                    "Hello {0} - Should I commit transaction {1}?", Name, txId), _
                    "Transaction", MessageBoxButtons.YesNo)
            Case DialogResult.Yes
                ContextUtil.SetComplete()
            Case DialogResult.No
                ContextUtil.SetAbort()
        End Select
    End Sub

    '
    ' TODO: Add RandomOutcome function
    '
End Class

File: GotDotNet.com/ES Hands On Lab/CS/Exercise 3/HelloEnterpriseServices/HelloTx.cs
using System;
using System.Windows.Forms;
using System.EnterpriseServices;
using System.Collections;
using System.Diagnostics;

namespace HelloEnterpriseServices
{
	[Transaction]
	public class HelloTx : ServicedComponent
	{
		public HelloTx()
		{
		}

		public void SayHello(string Name)
		{
			Guid txID = ContextUtil.TransactionId;
			string Msg = string.Format("Hello {0} - Commit transaction {1}?", Name, txID);
			switch (MessageBox.Show(Msg,"Transaction", MessageBoxButtons.YesNo))
			{
				case DialogResult.Yes:
					ContextUtil.SetComplete();
					break;
				case DialogResult.No:
					ContextUtil.SetAbort();
					break;
			}
		}

		private bool RandomCommit(out string Message)
		{
			bool Commit;
			Random r = new Random(DateTime.Now.Millisecond);
			// Commit if an even number is returned from the random generator
			Commit = ((r.Next() % 2) == 0) ? true : false;
			Message = String.Format("RandomOutcome Result: {0} Transaction ID: {1}", Commit ? "commit" : "abort", ContextUtil.TransactionId);
			return Commit;
		}

		//
		// TODO: Add RandomOutcome method
		//
	}
}
